### PR TITLE
Jsni feature

### PR DESCRIPTION
**Pull request by Chris2011**
related to this ticket: https://github.com/ksfreitas/gwt4nb/issues/36

### CODE REVIEW

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (reference == null) {
+            throw new NullPointerException(String.valueOf(errorMessage));
+        }
+
+        return reference;
+    }
+
+    @Override
+    public LanguageEmbedding<?> findLanguageEmbedding(Token<?> token, LanguagePath languagePath, InputAttributes inputAttributes) {
+        System.out.println(END_FRAGMENT);
```
Comment (markiewb) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java]: Remove all system.outs!

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/org-netbeans-modules-gwt4nb-HostedModeAction_1.shadow [hunk 2]
```diff
@@ -1 +0,0 @@
-file:/UNKNOWN
```
Comment (markiewb) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/org-netbeans-modules-gwt4nb-HostedModeAction_1.shadow]: Remove this file from comment

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    private Language embeddedLanguage;
+    public static final String START_FRAGMENT = "/*-{";
+    public static final String END_FRAGMENT = "}-*/";
+
+    @Override
+    public Language<?> findLanguage(String mimeType) {
+        return null;
+    }
+
+    private <T> T checkNotNull(T reference, Object errorMessage) {
```
Comment (markiewb) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java]: Change type of errorMessage to String

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.netbeans.api.lexer.LanguagePath;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.api.lexer.TokenUtilities;
+import org.netbeans.spi.lexer.LanguageEmbedding;
+import org.netbeans.spi.lexer.LanguageProvider;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Chrizzly
```
Comment (markiewb) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java]: Use your github-login

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java (hunk 1 above)
Comment (Chris2011) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java]: Sure, I forgot that.

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/org-netbeans-modules-gwt4nb-HostedModeAction_1.shadow (hunk 2 above)
Comment (Chris2011) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/org-netbeans-modules-gwt4nb-HostedModeAction_1.shadow]: Thx, I will.

File: branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+
+        return null;
+    }
+
+    private void initLanguage() {
+        embeddedLanguage = MimeLookup.getLookup("text/javascript").lookup(Language.class);
+
+//        checkNotNull(embeddedLanguage, "Can't find language for embedding");
```
Comment (markiewb) [inline branches/standalone_nbm_prj/src/org/netbeans/modules/gwt4nb/JSNISupport.java]: Remove comments

**Comment by markiewb** [comment]
BTW: Apply your changes to the ```trunk```. Not ```branches/standalone_nbm_prj```

**Comment by Chris2011** [comment]
I don't understand the project structure and I don't know why it was so complicated. Maybe it was a fork from a SVN repo or smth like that?

**Comment by markiewb** [comment]
@Chris2011: Yes it seems so. I has been a SVN project ages ago.

**Comment by Chris2011** [comment]
For now, I would like to leave it there, because it is only a "quick fix" with a lot of stuff, what is not working. Maybe after we can fix the main problem, wie can refactor the code.

**Comment by Chris2011** [comment]
Only my opinion.

**Comment by markiewb** [comment]
The main development happens on trunk. At least the last changes where there. 
So your changes will never get into the plugin, if you do not change it there.

**Comment by Chris2011** [comment]
I will have a look for the trunk.

**Comment by Chris2011** [comment]
Moved the changes to the trunk folder

**Comment by markiewb** [comment]
Fixed via https://github.com/ksfreitas/gwt4nb/commit/0546eb9e04a2cf5674b14ef1cde02e7ffe7bc581

**Pull request closed without merging**